crates/dslab-storage/src/fs.rs

```diff
@@ -250,38 +250,35 @@ impl FileSystem {
 
     /// Returns amount of used space on all disks currently mounted to this file system.
     pub fn used_space(&self) -> u64 {
-        self.disks.iter().map(|(_, v)| v.borrow().used_space()).sum()
+        self.disks.values().map(|v| v.borrow().used_space()).sum()
     }
 
     /// Returns amount of free space on all disks currently mounted to this file system.
     pub fn free_space(&self) -> u64 {
-        self.disks.iter().map(|(_, v)| v.borrow().free_space()).sum()
+        self.disks.values().map(|v| v.borrow().free_space()).sum()
     }
 
     /// Returns cumulative capacity of all disks currently mounted to this file system.
     pub fn capacity(&self) -> u64 {
-        self.disks.iter().map(|(_, v)| v.borrow().capacity()).sum()
+        self.disks.values().map(|v| v.borrow().capacity()).sum()
     }
 
     /// Returns vec of disk info associated with mount points.
-    pub fn disks_info(&self) -> Vec<(String, DiskInfo)> {
+    pub fn disks_info(&self) -> Vec<(String, StorageResourceInfo)> {
         self.disks
             .iter()
             .map(|(mount_point, disk)| (mount_point.to_owned(), disk.borrow().info()))
             .collect()
     }
 
     /// Returns disk info for a mount point.
-    pub fn disk_info(&self, mount_point: &str) -> Result<DiskInfo, String> {
+    pub fn disk_info(&self, mount_point: &str) -> Result<StorageResourceInfo, String> {
         self.resolve_disk(mount_point).map(|disk| disk.borrow().info())
     }
 
     /// Returns mount points present in this file system.
     pub fn mount_points(&self) -> Vec<String> {
-        self.disks
-            .iter()
-            .map(|(mount_point, _)| mount_point.to_owned())
-            .collect()
+        self.disks.keys().map(|mount_point| mount_point.to_owned()).collect()
```

Comment:
Кажется это эквивалентно и заметно короче

```suggestion
        self.disks.keys().cloned().collect()
```